src/strutil.c

```diff
@@ -41,3 +42,142 @@ int str_ends(const char *str, const char *ends) {
 
   return str_eq(str + str_len - ends_len, ends);
 }
+
```

Comment:
Perhaps we should have a single-line comment above the new functions to describe what they do quickly.
I didn't remember what `strncasecmp` did, so I had to look it up. And [drumroll], it compares strings without looking at the casing.

How about something like this?
```
// Check if [str] starts with [starts] without case sensitivity
```

Reply:
Excellent, thanks!